src/CoreBundle/Controller/ChameleonController.php

```diff
@@ -542,7 +551,7 @@ public function PreOutputCallbackFunction(&$sPageContent)
                 $bHeaderParsed = true;
             }
         }
-        $sPageContent = $this->PreOutputCallbackFunctionReplaceCustomVars($sPageContent);
+        $sPageContent = $this->responseVariableReplacer->replaceVariables($sPageContent);
```

Comment:
`$this->responseVariableReplacer` can be `null` as it is injected via a setter injection.

Reply:
Then why isn't this dependency injected into the constructor?

Reply:
Subclasses rely on the constructor as-is and would break. This is how we solved it for every new dependency in recent times (and we discuss the topic every time ;-) ) - maybe we should find a better solution in general, but not in this PR.

Reply:
We should :)